File: backend/Entities/DailyTask.cs
using System;

namespace backend.Entities
{
    public partial class DailyTask
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public string Type { get; set; } = null!;
        public int? CreatedById { get; set; }
        public int ProdavnicaId { get; set; }
        public DateTime Date { get; set; }
        public string Status { get; set; } = null!;
        public DateTime? CompletedAt { get; set; }
        public int? CompletedById { get; set; }
        public string? CompletionNote { get; set; }
        public bool ImageAllowed { get; set; }
        public string? ImageAttachment { get; set; }
        public int? TemplateId { get; set; }
        public bool IsRecurring { get; set; }

        public virtual Korisnik? CreatedBy { get; set; }
        public virtual Korisnik? CompletedBy { get; set; }
        public virtual Prodavnica Prodavnica { get; set; } = null!;
        public virtual DailyTaskTemplate? Template { get; set; }
    }
}

File: auro/src/app/@core/data/daily-task.ts
export type DailyTaskType = 'REPETITIVE' | 'CUSTOM';
export type DailyTaskStatus = 'OPEN' | 'IN_PROGRESS' | 'DONE';

export interface DailyTask {
  id: number;
  title: string;
  description?: string;
  type: DailyTaskType;
  date: string;
  status: DailyTaskStatus;
  imageAllowed: boolean;
  imageAttachment?: string;
  completionNote?: string;
  completedAt?: string;
  completedBy?: string;
  createdBy?: string;
  prodavnicaId: number;
  prodavnicaBroj?: string;
  prodavnicaNaziv?: string;
  isEditable: boolean;
  isRecurring: boolean;
}

export interface DailyTaskStore {
  id: number;
  code: string;
  name: string;
  city?: string;
  format?: string;
  managerId?: number | null;
  managerName?: string | null;
}

export interface DailyTaskPayload {
  title: string;
  description?: string;
  date: string;
  imageAllowed: boolean;
  isRecurring: boolean;
}

export interface DailyTaskStatusPayload {
  status: DailyTaskStatus;
  completionNote?: string;
  image?: File | null;
  removeImage?: boolean;
}

export interface DailyTaskBulkPayload extends DailyTaskPayload {
  targetType: 'stores' | 'city' | 'manager' | 'format';
  storeIds?: number[];
  city?: string;
  managerId?: number;
  format?: string;
}

export interface DailyTaskBulkResult {
  brojZadataka: number;
  brojProdavnica: number;
}
